### ISSUE TITLE
renderer in settings file does not work

### ISSUE THREAD

**Issue by immetoo2** [openMSX/openMSX#1139]
When using the settings file to disable renderer it gets ignored
`<setting id="renderer">None</setting>`

**Comment by immetoo2** [comment]
yes and checked that from scripts command both none and None work.

**Comment by m9710797** [comment]
This is by design. It's to prevent the following scenario:
* User types the command `set renderer none` in the console.
* The video output (thus also the console) disappears.
* Users kills the openMSX process. (Because there's no way to normally close openMSX). This would save "renderer=none" in the settings.xml file.
* When openMSX is restarted no video output appears.

The only way to recover from this is to edit or delete the settings.xml file. To prevent this we don't save "renderer=none" in the settings.xml file. And when it does end up in there we ignore it and fall back to the default renderer.

If you want such tight control over how openMSX starts, then maybe it's a good idea to use the "-control stdio" command line option. See http://openmsx.org/manual/openmsx-control.html. This will start openMSX with the none-renderer (and MSX in power=off state).

**Comment by immetoo2** [comment]
Or a extra setting we can set like 'auto_renderer_fallback=false'

**Comment by m9710797** [comment]
Both suggestions would work. But I'm personally not in favor. IMHO openMSX is already too complex and this suggestions will add more special cases to an already exceptional case. I'm not sure it's possible, but I'd welcome suggestions that regularize the original special case (without re-opening the usability pitfall)

I'm not against extending openMSX. But in this specific case, is this worth it? Isn't passing `set renderer none` in a script good enough? Or alternatively start openMSX via `-control stdio`.

**Comment by immetoo2** [comment]
The renderer setting on bootup has 2 usecases;
- Saw scripts use it as bootup speed improvement
   (it looks like throttle off doesn't disables vdp throttle?)
- Allows for full headless operation.

I have it now in tcl script which gave nice improvement from 33s to 9s running openMSX 12 times.
But still have a flickering window on every boot which takes lots of time.

Don't want to go with stdio in build steps as that makes it slower and more complex.
Have been thinking about a solution which doesn't add more complex path and works with original case, Can we override the default renderer with env var like DEFAULT_RENDERER=none ?

**Comment by m9710797** [comment]
Is the following too complex?

Create a file `control.txt` containing:
`<openmsx-control>`
`<command>unset renderer</command>`
`<command>set power on</command>`

Then start openMSX like this:
`openmsx -control stdio < control.txt`

If you leave out the 2nd line (`unset renderer`, meaning restore the `renderer` setting to its default value), then you have a headless openMSX instance.

You can combine this with the `-script` option or you can merge all commands of your script into `control.txt`. Alternatively you can have `control.txt` only contain the first line (`<openmsx-control>`) and move the `set power on` command into the Tcl script.

Basically the thing we're exploiting here is the fact that `-control` causes:
* openMSX to start with `renderer=none`
* openMSX doesn't do an automatic `set power on`.

As a bonus you get to see the output of each executed command on stdout (each command that was given via control.txt).  (If you don't want that, redirect stdout to `> /dev/null`).

**Comment by immetoo2** [comment]
I'll try it now

**Comment by immetoo2** [comment]
Yes it works and saves almost 2secs here.
Was checking what outputs on stdout and stderr now have
```
<openmsx-output>
<reply result="ok">true</reply>
<log level="info">Screen saved to bin/dist_run.png</log>
error: Failure request from dist_run
</openmsx-output>
```

At least all info is there, is there an option to runtime disable the `-control` mode so that stdout goes back to none-xml output ?

**Comment by immetoo2** [comment]
Have added `| sed -n -e 's/.*>\(.*\)<.*/\1/p'` to restore stdout.

And replaced my `set renderer SDL` with `unset renderer` to set the default renderer in autoexec.bat when needed.

And boot xml may be terminated so valid xml file can be used
```
<openmsx-control>
<command>set power on</command>
</openmsx-control>
```

This works for me so closing.
